crit-go/images/images.go

```diff
@@ -0,0 +1,240 @@
+package images
+
+import (
+	"encoding/binary"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+)
+
+func check(e error) {
+	if e != nil {
+		fmt.Fprintln(os.Stderr, e)
+		os.Exit(1)
+	}
+}
+
+func checkfile(e error, f *os.File) {
+	if e != nil {
+		f.Close()
+		fmt.Fprintln(os.Stderr, e)
+		os.Exit(1)
+	}
+}
+
+const sizeof_u16 = 2
+const sizeof_u32 = 4
+const sizeof_u64 = 8
+
+var jsonmap map[string]interface{}
+var bynamemap map[string]interface{}
+var byvalmap map[string]interface{}
+
+func Load(f *os.File, pretty bool, nopl bool) map[string]interface{} {
+	image := make(map[string]interface{})
+	/*
+	   Convert criu image from binary format to dict(json).
+	   Takes a file-like object to read criu image from.
+	   Returns criu image in dict(json) format.
+	*/
+	magicjson()
+	m := magichandler(f)
+	image["magic"] = m
+	switch {
+	case m == "GHOST_FILE":
```

Comment:
we can have a map of handlers and the code will looks like this:
handler, ok := handlers[m];
if !ok {
  		handler := &entry_handler{}
}
handler.m = m
image["entries"] = handler.Load(f, pretty, nopl)

Reply:
this will require type assertion for the struct methods like load etc to work which basically circles back to a type switch.